agent/util-scripts/pbench-postprocess-tools

```diff
@@ -82,37 +82,55 @@ if [[ ! -d "${tool_output_dir}" ]]; then
 	exit 1
 fi
 
+tmp=${pbench_tmp}/pbench-postprocess-tools.$$
+trap "rm ${tmp}" EXIT QUIT INT
+pbench-list-tools --group ${group} --with-option > ${tmp}
+
+# function to parse the above output
+function get_host {
+    # parse a hostentry and return the host
+    IFS=';' read -a hosttools <<<${1}
+    IFS=':' read -a host <<<${hosttools[1]}
+    echo ${host[1]}
+}
+
+function get_tools_entry {
+    # parse a host entry and return the tools
+    IFS=';' read -a hosttools <<<${1}
+    IFS=':' read -a itools <<<${hosttools[2]}
+    echo ${itools[1]}
+}
+
+function get_tools {
+    # parse a tools entry into separate tool and options
+    IFS=',' read -a otools <<<${1}
+    for tool in ${otools[@]} ;do
+        IFS=' ' read -a atool <<<${tool}
+        tools[${atool[0]}]=${atool[@]:1}
+    done
+}
+
 let failures=0
-for dirent in $(/bin/ls -1 ${tool_group_dir}); do
-        if [[ "${dirent}" == "__trigger__" ]]; then
-		# Ignore trigger files
-		continue
-	elif [[ ! -d ${tool_group_dir}/${dirent} ]]; then
-		# Skip spurious files of ${tool_group_dir}
-		warn_log "[${script_name}] \"${this_tool_file}\" is a file in \"${tool_group_dir}\"; that should not happen. Please consider deleting it."
-		continue
-	fi
+while read hostentry ;do
+	host=$(get_host ${hostentry})
+	tools_entry=$(get_tools_entry ${hostentry})
+	declare -A tools=()
+	get_tools ${tools_entry}
+
 	# FIXME: add support for label applied to the hostname directory.
-	host_tool_output_dir="${tool_output_dir}/${dirent}"
+	host_tool_output_dir="${tool_output_dir}/${host}"
```

Comment:
AFAICT, the original uses hard-tabs  so I changed the new version to use hard-tabs as well. `git diff` and `cat -T` show both files using hard-tabs. I changed the mode line to use a tab-width of 4 and also the basic offset and indentation to 4 as well.  That should make it consistent - I hope. I could turn off `indent-tabs-mode` and have everything be spaces, but the diff would show whitespace changes.

Reply:
As I said, I think we're supposed to use spaces, not tabs, so, at least for the lines that you've changed and the lines that you've added, you should probably not use tabs.  (I won't object if you fix the entire file, but I also won't object if you opt not to -- there is a tick-box in GitHub's difference settings to ignore whitespace changes, so that shouldn't be a problem.)

Reply:
I have some cleanups to do on various things, so let me do the tabs-to-spaces conversion in a separate PR.